wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/drivedistanceoffboard/subsystems/DriveSubsystem.java

```diff
@@ -114,4 +117,72 @@ public void resetEncoders() {
   public void setMaxOutput(double maxOutput) {
     m_drive.setMaxOutput(maxOutput);
   }
+
+  /**
+   * Creates a command to drive forward a specified distance using a motion profile.
+   *
+   * @param distance The distance to drive forward.
+   * @return A command.
+   */
+  public Command profiledDriveDistance(double distance) {
+    var profile =
+        new TrapezoidProfile(
+            new TrapezoidProfile.Constraints(
+                DriveConstants.kMaxSpeedMetersPerSecond,
+                DriveConstants.kMaxAccelerationMetersPerSecondSquared));
+    var timer = new Timer();
+    return startRun(
+            () -> {
+              // Restart timer so profile setpoints start at the beginning
+              timer.restart();
+              resetEncoders();
+            },
+            () -> {
+              // Current state never changes, so we need to use a timer to get the setpoints we need
+              // to be at
+              var setpoint = profile.calculate(timer.get(), new State(), new State(3, 0));
+              setDriveStates(setpoint, setpoint);
+            })
+        .until(() -> profile.isFinished(0));
```

Comment:
Is using the timer for `profile.calculate()` correct?

Reply:
Yes. Since the current state never changes, the time must advance to get the next setpoint (as opposed to using the previously calculated state as the current state, and using the loop period for the time).

Reply:
Oops, just realized there's literally a comment in there explaining that...